addon/controllers/pagination.js

```diff
@@ -0,0 +1,9 @@
+import Controller from "@ember/controller";
+import { tracked } from "@glimmer/tracking";
+
+export default class PaginationController extends Controller {
+  queryParams = ["page", "search"];
+
+  @tracked page = 1;
+  @tracked search;
```

Comment:
the `|| null` is primarily so that a queryParam which is set to `""` is not serialized. i changed it so that only strings are checked since this approach could cause problems with falsy values.